LocalPackages/SyncUI/Sources/SyncUI/Views/ScanOrPasteCodeView.swift

```diff
@@ -0,0 +1,255 @@
+                    .onTapGesture {
+                        _ = model.codeScanned("camera unavailable")
+                    }
+
+                Text(UserText.cameraIsUnavailableTitle)
+                    .font(.system(size: 20, weight: .bold))
+                    .padding(.bottom, 8)
+
+                Text(UserText.cameraIsUnavailableMessage)
+                    .lineLimit(nil)
+                    .multilineTextAlignment(.center)
+                    .font(.system(size: 16, weight: .regular))
+
+                Spacer()
+            }
+            .padding(.horizontal, Constants.hPad)
+        }
+    }
+
+    @ViewBuilder
+    func instructions() -> some View {
+
+        Text(model.isInRecoveryMode ? UserText.recoveryModeInstructions : UserText.connectDeviceInstructions)
+            .lineLimit(nil)
+            .multilineTextAlignment(.center)
+            .font(.system(size: 16, weight: .regular))
+            .padding(.vertical)
+
+    }
+
+    @ViewBuilder
+    func buttons() -> some View {
+
+        Section {
+            Group {
+                NavigationLink {
+                    PasteCodeView(model: model)
+                } label: {
+                    Label(UserText.manuallyEnterCodeLabel, image: "SyncKeyboardIcon")
+                }
+
+                if !model.isInRecoveryMode {
+                    NavigationLink {
+                        ConnectModeView(model: model)
+                    } label: {
+                        Label(UserText.showQRCodeLabel, image: "SyncQRCodeIcon")
+                    }
+                }
+            }
+            .frame(height: 40)
+            .foregroundColor(.primary)
+        }
+
+    }
+
+    @ViewBuilder
+    func cameraViewPort() -> some View {
+        ZStack(alignment: .center) {
+            waitingForCameraPermission()
+            cameraTarget()
+        }
+    }
+
+    @ViewBuilder
+    func cameraTarget() -> some View {
+        if model.showCamera {
+            ZStack {
+                ForEach([0.0, 90.0, 180.0, 270.0], id: \.self) { degrees in
+                    RoundedCorner()
+                        .stroke(lineWidth: 8)
+                        .foregroundColor(.white.opacity(0.8))
+                        .rotationEffect(.degrees(degrees), anchor: .center)
+                        .frame(width: 300, height: 300)
+                }
+            }
+        }
+    }
+
+    public var body: some View {
+        GeometryReader { g in
+            ZStack(alignment: .top) {
+                fullscreenCameraBackground()
+
+                VStack(spacing: 0) {
+                    Rectangle() // Also acts as the blur for the camera
+                        .fill(.clear)
+                        .frame(height: g.safeAreaInsets.top)
+                        .regularMaterialBackground()
+
+                    ZStack {
+                        // Background in case fullscreen camera view doesn't work
+                        Rectangle().fill(Color.black)
```

Comment:
This covers the actual camera viewport (also when the camera is fully functional – I can scan a QR code in this state):
![IMG_9827](https://user-images.githubusercontent.com/14022/222403756-1f53f77f-d2ad-4567-a1c2-36593caa88f8.PNG)